### PR TITLE
Adds new feature: New button in the New and Old UI Diff Viewers to Expand All Diff Sections

### PR DESCRIPTION
**Pull request by RobertChrist**
-   [ ] I updated the CHANGELOG.md
-   [x] I tested the changes in this pull request myself
-   [x] I added Automated Tests
-   [x] I added an Option to enable / disable this feature
-   [x] I updated the README.md, with pictures if necessary

---

This PR adds a new button in the new and old UI diff viewers, next to the copy filename button, that when clicked, will click all of the individual "..." buttons in the diff.  So if your diff has 4 expandable "..." sections, clicking this button will save you 3 clicks.  

This is particularly useful for large files with lots of small edits, as those files tend to result in very disjointed and hard to read diffs, until you have expanded the diff enough to view a significant portion of the file.

<details>
  <summary>See GIF</summary>

gif

</details>

### CODE REVIEW

File: src/insert-expand-diff/insert-expand-diff.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    const header: ?HTMLElement = diff.querySelector('.filename')
+
+    if (!header) return
+
+    const lozenge: ?HTMLElement = header.querySelector('h1 > :last-child')
+
+    if (!lozenge) return
+
+    header.insertBefore(button, lozenge)
```
Comment (reyronald) [inline src/insert-expand-diff/insert-expand-diff.js:60]: What if we don't render the button if there's no `ellipsis` /  `Show more lines button` since it'll serve no purpose ? Or maybe render it in disabled state ? Thoughts?

Same question for the new UI

File: src/insert-expand-diff/insert-expand-diff.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export default function insertExpandDiff(diff: HTMLElement) {
+    const button = (
+        <button
+            type="button"
+            class="aui-button aui-button-subtle devtools-task-in-progress--button __rbb-btn-expandDiff"
+            title="Expand unexpanded diff sections once"
+            original-title="Expand unexpanded diff sections once"
+            style={{ position: 'relative' }}
```
Comment (reyronald) [inline src/insert-expand-diff/insert-expand-diff.js]: How would you feel if we add some styles here to make the two buttons look more compact ? Right now there's a lot of extra spacing on the new one that makes it look asymmetric

Now

image

After

image

Styles I used to take the second screenshot:

```css
    margin: 0;
    padding-left: 5px;
    padding-right: 5px;

```

It looks great in the new UI though!

File: src/insert-expand-diff/insert-expand-diff.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    if (!expandElipses) return
+
+    expandElipses.forEach(expander => expander.click())
+}
+
+export default function insertExpandDiff(diff: HTMLElement) {
+    const button = (
+        <button
+            type="button"
+            class="aui-button aui-button-subtle devtools-task-in-progress--button __rbb-btn-expandDiff"
```
Comment (reyronald) [inline src/insert-expand-diff/insert-expand-diff.js:33]: Is this `devtools-task-in-progress--button` CSS class available in Firefox? Does the button look as intended in Firefox ?

File: src/insert-copy-filename/insert-copy-filename.js [hunk 4]
```diff
@@ -47,6 +47,6 @@ export default function insertCopyFilename(diff: HTMLElement) {
     )
 
     const header: HTMLElement = (diff.querySelector('.filename'): any)
-    const lozenge: HTMLElement = (header.querySelector(':last-child'): any)
+    const lozenge: HTMLElement = (header.querySelector('h1 > :last-child'): any)
```
Comment (reyronald) [inline src/insert-copy-filename/insert-copy-filename.js:50]: Why this change ?

**Review by reyronald — COMMENTED** [review]
This looks great Robert thanks a lot! Left a few minor questions

File: src/insert-expand-diff/insert-expand-diff.js (hunk 1 above)
Comment (RobertChrist) [reply src/insert-expand-diff/insert-expand-diff.js:60]: Makes complete sense to me.  I'm quite busy this week, but will attempt to patch this idea in shortly : )

File: src/insert-expand-diff/insert-expand-diff.js (hunk 2 above)
Comment (RobertChrist) [reply src/insert-expand-diff/insert-expand-diff.js]: makes sense!

File: src/insert-expand-diff/insert-expand-diff.js (hunk 3 above)
Comment (RobertChrist) [reply src/insert-expand-diff/insert-expand-diff.js:33]: I tested this in chrome and firefox, and it looked fine in both.  The truth is I got these classnames from the atlassian documentation recommended when reusing their imageset.  I've attempted to find this exact string again, and cannot find it anywhere on any atlassian documentation page.  Atlassian documentation isn't the easiest thing to navigate on the best of days, but I think it might have genuinely changed since last October, regardless.

File: src/insert-copy-filename/insert-copy-filename.js (hunk 4 above)
Comment (RobertChrist) [reply src/insert-copy-filename/insert-copy-filename.js:50]: When I was testing this last November, I found that with both the insert-copy-filename feature and this PR's feature enabled, the insert-copy-filename would assume that its button was the last one in the DOM, and could end up accidentally selecting the HTML for my button instead, causing issues.  This small selector change resolved that.

I attempted to test this a few weeks ago, and could not replicate the issue, but would need to do more testing to feel confident one way or the other about removal.

**Comment by RobertChrist** [comment]
I'm choosing to close this pull request.  I have been unable to find the time to devote to pushing this feature over the finish line, which means I am also unlikely to have the time to maintain it going forward.  In the event that I get enough time to implement this feature in the future, I will reopen this pr.
